src/test/java/org/broadinstitute/hellbender/IntelDeflaterIntegrationTest.java

```diff
@@ -1,29 +1,129 @@
 
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
 /**
  * Test that it's possible to load libIntelDeflater
  */
 public class IntelDeflaterIntegrationTest extends BaseTest {
 
-    @Test
-    public void testIntelDeflaterIsAvailable(){
-        if ( ! NativeUtils.runningOnLinux() ) {
-            throw new SkipException("IntelDeflater not available on this platform");
-        }
+    private final static Logger log = LogManager.getLogger(IntelDeflaterIntegrationTest.class);
+    private static final String publicTestDirRelative = "src/test/resources/";
+    private static final String publicTestDir = new File(gatkDirectory, publicTestDirRelative).getAbsolutePath() + "/";
+    private static final String INPUT_FILE = publicTestDir + "CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam";
 
-        if ( NativeUtils.runningOnPPCArchitecture() ) {
-            throw new SkipException("IntelDeflater not available for this architecture");
-        }
+    @Test(enabled = true)
+    public void loadIntelDeflater() {
+        // create buffers and random input
+        final int LEN = 64 * 1024;
+        final byte[] input = new RandomDNA().nextBases(LEN);
+        final byte[] compressed = new byte[2 * LEN];
+        final byte[] result = new byte[LEN];
+
+        final IntelDeflaterFactory intelDeflaterFactory = new IntelDeflaterFactory();
+
+        for (int i = 0; i < 10; i++) {
+            // create deflater with compression level i
+            final Deflater deflater = intelDeflaterFactory.makeDeflater(i, true);
+            Assert.assertTrue(intelDeflaterFactory.usingIntelDeflater());
+
+            // setup deflater
+            deflater.reset();
+            deflater.setInput(input);
+            deflater.finish();
 
-        Assert.assertTrue(DeflaterFactory.usingIntelDeflater(), "libIntelDeflater.so was not loaded. " +
-                "This could be due to a configuration error, or your system might not support it.");
+            // compress data
+            int compressedBytes = 0;
+            while (!deflater.finished()) {
+                compressedBytes = deflater.deflate(compressed, 0, compressed.length);
+            }
+            deflater.end();
+
+            // log results
+            log.info("%d bytes compressed to %d bytes : %2.2f%% compression\n",
+                     LEN, compressedBytes, 100.0 - 100.0 * compressedBytes / LEN);
+
+            // decompress and check output == input
+            Inflater inflater = new Inflater(true);
+            try {
+                inflater.setInput(compressed, 0, compressedBytes);
+                inflater.inflate(result);
+                inflater.end();
+            } catch (java.util.zip.DataFormatException e) {
+                e.printStackTrace();
+            }
+
+            Assert.assertEquals(input, result);
+        }
     }
 
+    @Test(enabled = false)
```

Comment:
The test uses `SAMFileWriterFactory.setDeflaterFactory`, which isn't available in a released version of HTSJDK.